src/components/ShareButton.jsx

```diff
@@ -2,17 +2,76 @@ import { Button, Snackbar } from "@mui/material";
 import IosShareIcon from "@mui/icons-material/IosShare";
 import MuiAlert from "@mui/material/Alert";
 import React from "react";
+import { useFormContext } from "context/Form";
+import { fetchClasses } from "forms/Schedule";
+
+const rootURL = `${window.location.protocol}//${window.location.hostname}${
+  window.location.port ? ":" + window.location.port : ""
+}`;
+
+export const generateShareLink = async ({
```

Comment:
All the code in `generateShareLink` is synchronous, references to `generateShareLink` have an await in front. I suggest dropping the async and awaits

Reply:
I missed the fetch going on in here, async is 👍